src/core/backend/pearl/group.rs

```diff
@@ -121,32 +123,40 @@ impl Group {
     }
 
     // create pearl for current write
-    async fn create_write_pearl(&self, timestamp: u64) -> BackendResult<Holder> {
-        self.settings
-            .config()
-            .try_multiple_times_async(
-                || self.try_create_write_pearl(timestamp),
-                "pearl init failed",
-                self.settings.config().settings().create_pearl_wait_delay(),
-            )
+    async fn create_write_pearl(&self, ts: u64) -> BackendResult<Holder> {
+        if !self.created_holder_indexes.read().await.contains_key(&ts) {
+            let mut indexes = self.created_holder_indexes.write().await;
```

Comment:
Could it be here a Deadlock if you take read lock right before?

Reply:
Readlock gets dropped, isn't it?

Reply:
@pyakushin what do you think.
@idruzhitskiy what about race condtion?